### ISSUE TITLE
CompletedRequest.body.getFormData() returns very raw form data 

### ISSUE THREAD

**Issue by exoego** [httptoolkit/mockttp#157]
- mockttp v3.9.4
- Node.js v18.16.0

Given
```js
const mockServer = getLocal();
mockServer.start(18080);
await mockServer.forPost("/user").thenCallback(async (req) => {
    const formData = await req.body.getFormData();
    return {
        statusCode: 200,
        json: formData,
    };
});

const formData = new FormData();
formData.append("name", "John Doe");

const response = await fetch("http://localhost:18080/user", {
    method: "POST",
    body: formData,
});
```

I expect `response`, the result of `getFormData()`, to be:
```json
{
  "name": "John Doe"
}
```

But I get a weird JSON:

```json
{
 "------formdata-undici-055913559201\r\n Content-Disposition: form-data; name": 
    "\"name\"\r\n\r\n John Doe\r\n ------formdata-undici-055913559201--"
}
```

Is this intended behavior??

**Comment by pimterry** [comment]
Hmm, interesting. `getFormData()` is intended to parse `application/x-www-form-urlencoded` format data (which looks like `a=1&b=2&...`) which is the standard format that data is sent by HTML web forms.

This data is not that. Instead this is `multipart/form-data` data. In a web form, that's normally only used if there's a file upload involved. Clearly it's the default for `fetch` with FormData though, which is news to me.

MDN has an example of multipart data [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types#multipartform-data) - basically there's a header like `Content-Type: multipart/form-data; boundary=aBoundaryString`, and then a series of sections in the body separated with `--aBoundaryString` (two dashes before the given boundary) and their own set of headers and body (with a blank line between). For form data, the part headers give you the name of the form field (`Content-Disposition: form-data; name="your field name"`) and the whole body is the value.

We could definitely parse this too, and I think it'd make sense to do so. Would you be up for opening a PR to do that?

The relevant code is here: https://github.com/httptoolkit/mockttp/blob/1106f4aa6963f3822408605da78d79e8be9eed99/src/util/request-utils.ts#L238-L243

You'd need to extend that to:

* Check the content-type, and get the `boundary` value if it's multipart form data
* Parse the body using that if so
* If not, parse the body as URL encoded data, just like we do today (to avoid breaking those cases)

We actually already support multipart form data for matching ([here](https://github.com/httptoolkit/mockttp/blob/1106f4aa6963f3822408605da78d79e8be9eed99/src/rules/matchers.ts#L385-L421)) so that code could be reused to handle most of this.

I think in future the API should probably be more explicit about the two formats, but we could keep this method to handle both, so we'd have `getUrlEncodedData()` with the current logic, `getMultipartFormData()` with this new logic, and `getFormData()` that automatically uses one of those depending on the content type. We can do that separately though.

**Comment by pimterry** [comment]
Oh btw, in case you're not aware, all contributors to everything in this org get free [HTTP Toolkit Pro](https://httptoolkit.com).

**Comment by pimterry** [comment]
Now released as part of Mockttp v3.10.0.
